muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java

```diff
@@ -1337,6 +1344,15 @@ public Response searchStaffMembers(
             }
           }
           
+          // #3111: Workspace staff members should be limited to teachers only. A better implementation would support specified workspace roles
+          
+          if (workspaceEntity != null) {
```

Comment:
This method is used to search for staff members in general or staff members of a single workspace. In the latter case, each user's workspace role has to be manually checked as the user's role in each workspace they belong to is not stored in the Elastic index.

We're essentially trying to figure out whether the staff member belonging to a workspace is actually a teacher in that workspace. If so, they are included in the final result set. If they are, for example, just workspace tutors, they're not included.

Granted, the aforementioned logic should not even reside within a generic search function like this. Still, this was the simplest way to only show workspace teachers on the front page of a workspace.